### ISSUE TITLE
pointers as labels

### ISSUE THREAD

**Issue by endreszabo** [genelet/determined#2]
Hi Peter,

My goal is to render [Nomad job specification](https://developer.hashicorp.com/nomad/docs/job-specification]) HCL files based on the original [Nomad structs](https://github.com/hashicorp/nomad/blob/7b9bce2d083b82222086553b65bd40e750d00d8e/api/jobs.go#L960). In these structs, some labels are strings, and some are pointers to strings. In the final HCL, the label where the value is a pointer, the attribute gets rendered as a simple attribute and not as a block label.

Check the boilerplate code producing the sample output here: https://go.dev/play/p/f6WFxX91AED

Its the expected output is:

```
it works here:
  job "jobNameAsNormal" {

  }

it fails here:
  job {
    name  = jobNameAsPointer
  }
```

I might be in the wrong here, but I'm not sure, tbh.

Thanks,

--Endre

**Comment by genelet** [comment]
Hi, Endre:

The label - map key issue is fixed in the latest v1.11.1. Please check.

Peter

**Comment by genelet** [comment]
Hi, Endre:

1) There used to be a bug in the **Unmarshal** function that did not treat label and key properly. The latest version v1.11.1 fixed it.
2) The **Marshal** function does not follow tags in a struct. It simply marshals a map key as key in "block", and a tagged label field as "attribute" in HCL. However, when unmarshaling the HCL back, one gets consistently the same result. In your example, you may print  _jobList.Jobs[0].Name_ in the "normal" struct, and _*(jobList.Jobs[0].Name)_ in the "pointer" struct. They output same result, "jobNameAsNormal".

Hopefully this answers your question.
